### PR TITLE
Remove full DSNs from exception messages

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Follow up of #49065

We should be careful when reviewing notifier bridges, to ensure that transports' `__toString()` don't embed any secrets. /cc @OskarStark

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/AbstractAdapter.php [hunk 1]
```diff
@@ -126,7 +126,7 @@ public static function createConnection(#[\SensitiveParameter] string $dsn, arra
             return CouchbaseCollectionAdapter::createConnection($dsn, $options);
         }
 
-        throw new InvalidArgumentException(sprintf('Unsupported DSN: "%s".', $dsn));
+        throw new InvalidArgumentException('Unsupported DSN: it does not start with "redis[s]:", "memcached:" nor ""couchbase:".');
```
Comment (fancyweb) [inline src/Symfony/Component/Cache/Adapter/AbstractAdapter.php]:
```suggestion
        throw new InvalidArgumentException('Unsupported DSN: it does not start with "redis[s]:", "memcached:" nor "couchbase:".');
```

Comment (nicolas-grekas) [reply src/Symfony/Component/Cache/Adapter/AbstractAdapter.php]: fixed thanks

**Comment by NicolasCARPi** [comment]
Thanks everyone :+1: :)

File: src/Symfony/Component/Cache/Traits/RedisTrait.php [hunk 2]
```diff
@@ -160,17 +160,17 @@ public static function createConnection(#[\SensitiveParameter] string $dsn, arra
...
         $params += $query + $options + self::$defaultConnectionOptions;
 
         if (isset($params['redis_sentinel']) && !class_exists(\Predis\Client::class) && !class_exists(\RedisSentinel::class)) {
-            throw new CacheException(sprintf('Redis Sentinel support requires the "predis/predis" package or the "redis" extension v5.2 or higher: "%s".', $dsn));
+            throw new CacheException('Redis Sentinel support requires the "predis/predis" package or the "redis" extension v5.2 or higher.');
         }
 
         if ($params['redis_cluster'] && isset($params['redis_sentinel'])) {
-            throw new InvalidArgumentException(sprintf('Cannot use both "redis_cluster" and "redis_sentinel" at the same time: "%s".', $dsn));
+            throw new InvalidArgumentException('Cannot use both "redis_cluster" and "redis_sentinel" at the same time.');
```
Comment (alexisLefebvre) [inline src/Symfony/Component/Cache/Traits/RedisTrait.php:173]: `: ` has been changed to `.`, it looks like it breaks some tests: https://github.com/symfony/symfony/actions/runs/3988999209/jobs/6840883282#step:11:78

Comment (alexisLefebvre) [reply src/Symfony/Component/Cache/Traits/RedisTrait.php:173]: Let's fix this in GH-49080

**Comment by OskarStark** [comment]
I agree

**Pull request merged**